metrics/ci_integrations/test/client/buildkite/test_utils/mock/buildkite_mock_server.dart

```diff
@@ -231,35 +234,54 @@ class BuildkiteMockServer extends ApiMockServer {
     return int.tryParse(buildNumber);
   }
 
-  /// Returns the [BuildkiteBuildState], based on the `state` query parameter
-  /// of the given [request].
-  BuildkiteBuildState _extractBuildState(HttpRequest request) {
-    final state = request.uri.queryParameters['state'];
+  /// Returns a [List] of [BuildkiteBuildState]s, based on the `state` query
+  /// parameter of the given [request].
+  List<BuildkiteBuildState> _extractBuildStates(HttpRequest request) {
+    final requestedStates = request.uri.queryParametersAll['state[]'];
+
+    if (requestedStates == null) return null;
+
+    const buildkiteStateMapper = BuildkiteBuildStateMapper();
 
-    return const BuildkiteBuildStateMapper().map(state);
+    return requestedStates
+        .map((state) => buildkiteStateMapper.map(state))
+        .toList();
   }
 
-  /// Generates a list of [BuildkiteBuild]s with the given [state]
-  /// and [finishedFrom].
+  /// Generates a list of [BuildkiteBuild]s with the [BuildkiteBuildState]s
+  /// from the given [states] and [finishedFrom].
   ///
-  /// If the given [state] is `null`, the [BuildkiteBuildState.finished] is used.
   /// If the given [finishedFrom] is `null`, the [DateTime.now] is used.
+  /// Delegates selecting a random [BuildkiteBuildState]s from the given [states]
+  /// to the [_selectRandomState] method.
   List<BuildkiteBuild> _generateBuilds([
-    BuildkiteBuildState state,
+    List<BuildkiteBuildState> states,
     DateTime finishedFrom,
   ]) {
-    return List.generate(
-      100,
-      (index) => BuildkiteBuild(
+    return List.generate(100, (index) {
+      return BuildkiteBuild(
         id: 'id',
         number: index,
         blocked: false,
-        state: state ?? BuildkiteBuildState.finished,
+        state: _selectRandomState(states),
         webUrl: 'url',
         startedAt: DateTime(2020),
         finishedAt: finishedFrom ?? DateTime.now(),
-      ),
-    );
+      );
+    });
+  }
+
+  /// Selects a random [BuildkiteBuildState] from the given [states].
+  ///
+  /// Returns a [BuildkiteBuildState.passed] if the given [states] is `null` or
+  /// [List.isEmpty]. Otherwise, returns a random [BuildkiteBuildState] selected
+  /// from the given [states].
+  BuildkiteBuildState _selectRandomState([List<BuildkiteBuildState> states]) {
+    if (states == null || states.isEmpty) return BuildkiteBuildState.passed;
+
+    final randomStateIndex = _random.nextInt(states.length);
```

Comment:
```suggestion
    final stateIndex = buildNumber % states.length;
```
Randomness is not the best choice for tests as it may cause tests flakiness. Instead, we can select a state using modulo of states list length (more about why this should work: https://en.wikipedia.org/wiki/Modular_arithmetic#Integers_modulo_n).